### PR TITLE
contrib: use a distinct --pull-option= for each flag

### PR DESCRIPTION
**Pull request by giuseppe**
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: contrib/cirrus/setup_environment.sh [hunk 1]
```diff
@@ -57,8 +57,8 @@ if [[ "$CI_DESIRED_STORAGE" = "composefs" ]]; then
         # KLUDGE ALERT! Magic options needed for testing composefs.
         # This option was intended for passing one arg to --storage-opt
         # but we're hijacking it to pass an extra option+arg. And it
-        # actually works. Just MAKE SURE THERE ARE NO SPACES IN THE {...}!
-        export STORAGE_OPTIONS_OVERLAY='overlay.use_composefs=true --pull-option {enable_partial_images="true",use_hard_links="false",ostree_repos="",convert_images="true"}'
+        # actually works.
+        export STORAGE_OPTIONS_OVERLAY='overlay.use_composefs=true --pull-option=enable_partial_images=true --pull-option=convert_images=true'
```
Comment (rhatdan) [inline contrib/cirrus/setup_environment.sh:61]: Does containers/storage now understand --pull-option in environment variable?

**Comment by Luap99** [comment]
```
level=warning msg="Execution of `modprobe overlay` ended with error: exit status 1"
```

From rootless system tests

**Comment by giuseppe** [comment]
caused by https://github.com/containers/storage/commit/48e010f485ffe0dabc8a4e6092dfba2fc74f59c3

@Honny1 can you please lower that to a debug message?

**Comment by giuseppe** [comment]
> caused by [containers/storage@48e010f](https://github.com/containers/storage/commit/48e010f485ffe0dabc8a4e6092dfba2fc74f59c3)
> 
> @Honny1 can you please lower that to a debug message?

opened a PR https://github.com/containers/storage/pull/2019

**Comment by giuseppe** [comment]
dropped the c/storage bump, I'll deal with that later

**Review by Luap99 — COMMENTED** [review]
LGTM, is there way for us to throw an error when invalid options are passed to prevent these kind of errors?

**Comment by giuseppe** [comment]
> LGTM, is there way for us to throw an error when invalid options are passed to prevent these kind of errors?

currently not, opened an issue to not forget about it: https://github.com/containers/storage/issues/2020

**Pull request merged**